src/monodroid/jni/embedded-assemblies.cc

```diff
@@ -400,6 +392,17 @@ EmbeddedAssemblies::gather_bundled_assemblies_from_apk (const char* apk, monodro
 				continue;
 			}
 
+			if (utils.ends_with (cur_entry_name, ".jm")) {
+				md_mmap_info map_info   = md_mmap_apk_file (fd, offset, info.uncompressed_size, cur_entry_name, apk);
+				add_type_mapping (&java_to_managed_maps, apk, cur_entry_name, (const char*)map_info.area);
+				continue;
+			}
+			if (utils.ends_with (cur_entry_name, ".mj")) {
+				md_mmap_info map_info   = md_mmap_apk_file (fd, offset, info.uncompressed_size, cur_entry_name, apk);
+				add_type_mapping (&managed_to_java_maps, apk, cur_entry_name, (const char*)map_info.area);
+				continue;
+			}
+
```

Comment:
@grendello is there something I can do to skip these two blocks when the app starts with the shared runtime being used?

Reply:
Yep, we can [add a flag here](https://github.com/xamarin/xamarin-android/blob/9887642623f6cea964e8cfe302bca83f6387d068/src/monodroid/jni/xamarin-app.h#L17), initialize it to some [default here](https://github.com/xamarin/xamarin-android/blob/9887642623f6cea964e8cfe302bca83f6387d068/src/monodroid/jni/application_dso_stub.cc#L15), add an appropriate [property here](https://github.com/xamarin/xamarin-android/blob/9887642623f6cea964e8cfe302bca83f6387d068/src/Xamarin.Android.Build.Tasks/Utilities/ApplicationConfigNativeAssemblyGenerator.cs#L7), write it out to the [assembly source here](https://github.com/xamarin/xamarin-android/blob/9887642623f6cea964e8cfe302bca83f6387d068/src/Xamarin.Android.Build.Tasks/Utilities/ApplicationConfigNativeAssemblyGenerator.cs#L42-L70) (the fields **must** be written in the same order they are declared in the `xamarin-app.h` header!), assign the [correct value here](https://github.com/xamarin/xamarin-android/blob/9887642623f6cea964e8cfe302bca83f6387d068/src/Xamarin.Android.Build.Tasks/Tasks/GeneratePackageManagerJava.cs#L259-L267) and then query it like so whenever you need to know its value:

```cpp
if (application_config.YourFlagName)
  ;
```

When adding the flag, please keep it close to the other boolean flags :)